core/src/View/Kokkos_ViewAlloc.hpp

```diff
@@ -139,19 +124,52 @@ struct ViewValueFunctor<DeviceType, ValueType, /*IsTrivial=*/false> {
     }
   }
 
+  // Shortcut for zero initialization
+  void zero_memset_implementation() {
+    uint64_t kpID = 0;
+    if (Kokkos::Profiling::profileLibraryLoaded()) {
+      // We are not really using parallel_for here but using beginParallelFor
+      // instead of begin_parallel_for (and adding "via memset") is the best
+      // we can do to indicate that this is not supposed to be tunable (and
+      // doesn't really execute a parallel_for).
+      Kokkos::Profiling::beginParallelFor(
+          "Kokkos::View::initialization [" + name + "] via memset",
+          Kokkos::Profiling::Experimental::device_id(space), &kpID);
+    }
+
+    (void)ZeroMemset(
+        space, Kokkos::View<ValueType*, typename DeviceType::memory_space,
+                            Kokkos::MemoryTraits<Kokkos::Unmanaged>>(ptr, n));
+
+    if (Kokkos::Profiling::profileLibraryLoaded()) {
+      Kokkos::Profiling::endParallelFor(kpID);
+    }
+    if (default_exec_space) {
+      space.fence("Kokkos::View::initialization via memset");
+    }
+  }
+
   void construct_shared_allocation() {
-    parallel_for_implementation<ConstructTag>();
+#ifndef KOKKOS_ARCH_A64FX
```

Comment:
```suggestion
// On A64FX memset seems to do the wrong thing with regards to first touch
// leading to the significant performance issues
#ifndef KOKKOS_ARCH_A64FX
```

I think we should keep the comment here, just to document why this is necessary